frontend/src/components/navbar/Navbar.tsx

```diff
@@ -2,8 +2,13 @@ import { Link } from "react-router-dom";
 import style from "./navbar.module.scss";
 import logo from "assets/logo.svg";
 import magnifierIcon from "assets/magnifier.svg";
+import { RootState } from "store/store";
+
+import { useSelector } from "react-redux";
```

Comment:
Please instead of using useDispatch and useSelector hooks from react-redux,  create your own useAppDispatch and useAppSelector hooks. like in example below

import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";
import type { RootState, AppDispatch } from "../store";

export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
export const selectSelft = (state: RootState) => state;